marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/ingestion/bulkIngester.sjs

```diff
@@ -29,6 +29,29 @@ const consts = require('/data-hub/5/impl/consts.sjs');
 const state = fn.head(xdmp.fromJSON(endpointState));
 
 const work = fn.head(xdmp.fromJSON(workUnit));
+let collections=[];
+if(work.collections !=null){
+  collections=work.collections.split(',');
+}
+
+const permissionsArray = []
+
+if(!work.permissions){
+  work.permissions='data-hub-common,read,data-hub-common,update'
+}
+
+const permissions = work.permissions.split(',');
+
+if((permissions.length%2)!= 0){
+  xdmp.log("Error: Unable to parse Permissions")
+  fn.error(null, "RESTAPI-SRVEXERR", Sequence.from([400, "Illegal Argument Exception", "Unable to parse permissions string, which must be a comma-separated list of role names and capabilities - i.e. role1,read,role2,update,role3,execute; string:" + permissions]));
```

Comment:
Use ds-utils.sjs - that has helper functions in it to simplify this - e.g. throwBadRequest would be appropriate here.